tracer/src/Datadog.Trace/Tagging/InferredProxyTags.cs

```diff
@@ -0,0 +1,31 @@
+// <copyright file="InferredProxyTags.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.SourceGenerators;
+
+namespace Datadog.Trace.Tagging;
+
+internal partial class InferredProxyTags : InstrumentationTags
```

Comment:
```suggestion
internal partial class InferredProxyTags : InstrumentationTags, IHasStatusCode
```

----

We have some helper extension methods that rely on tag types implementing `IHasStatusCode`. Consider reusing those extensions for setting status codes instead of doing it directly. They do some validation and check settings to determine which ones should be considered errors.
https://github.com/DataDog/dd-trace-dotnet/blob/4e98576a4f8edd19e1e691fbdb54f8a0eec58dfd/tracer/src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs#L91-L133

----

`IHasStatusCode` is also used by the normalizer, if we ever actually get to use that code 😅 
https://github.com/DataDog/dd-trace-dotnet/blob/4e98576a4f8edd19e1e691fbdb54f8a0eec58dfd/tracer/src/Datadog.Trace/Processors/NormalizerTraceProcessor.cs#L131

Reply:
Ahhhh thanks didn't realize that!